nicomakiumi

```diff
@@ -0,0 +1,618 @@
+            local myChoice  = _ZHINIAN_DRAW
+            if myThrow == nil then 
+            else
+                myChoice = _ZHINIAN_THROW
+            end            
+            local hisThrow = room:askForCard(current, ".|black|.", "zhinian_throw_prompt", data, sgs.Card_MethodNone) 
+            local hisChoice = _ZHINIAN_DRAW            
+            if hisThrow == nil then
+            else
+                hisChoice = _ZHINIAN_THROW
+            end            
+            if myChoice ~= hisChoice then
+                print("What a pity! We've made different choices!")
+                return false
+            end
+            
+            print("Good! We made the same choice")
+            if myChoice == _ZHINIAN_DRAW then
+                player:drawCards(1)
+                current:drawCards(1)
+            else
+                room:throwCard(myThrow, player)
+                room:throwCard(hisThrow, current)                
+            end
+            room:setPlayerFlag(current, "zhinian_donar")
+            room:setPlayerFlag(player, "zhinian_acceptor")            
+
+        elseif event == sgs.EventPhaseChanging then 
+            -- if not room:getTag("zhinian_extra") then
+            local zhinian_extra = room:getTag("zhinian_extra")
+            if zhinian_extra:toInt() == 0 then
+                --print("This is a regular turn")
+                if data:toPhaseChange().to ~= sgs.Player_NotActive or 
+                    not player:hasFlag("zhinian_donar") then
+                    return false
+                end
+                local nico = room:findPlayerBySkillName(self:objectName())
+                if nico == nil or not nico:hasFlag("zhinian_acceptor") then
+                    return false
+                end
+                if not room:askForSkillInvoke(nico, self:objectName(), data) then
+                    print("Nico does not want another phase of play")
+                    return false
+                end
+                print("Nico chooses to gain an extra phase of play")
+                -- Cannot simply gain another phase of play
+                -- Instead, gain an extra turn and skip all phases except play
+                room:setTag("zhinian_extra", sgs.QVariant(1))
+                nico:gainAnExtraTurn()
+                print("Extra turn finished")
+                room:removeTag("zhinian_extra")
+            else
+                local change = data:toPhaseChange()
+                if change.to ~= sgs.Player_Play then
+                    player:skip(change.to)                
+                end
+                
+            end
+        end
+    end,
+    
+    can_trigger = function(self, target)
+		return target
+	end
+}
+
+
+LuawudaoCard = sgs.CreateSkillCard{
+	 name = "LuawudaoCard",
+     target_fixed = true,
+     will_throw = false,
+	 
+	 on_use = function(self, room, source)
+		if not source:isKongcheng() then
+					local card_id = -1
```

Comment:
As you can see, though Tab is usually displayed as 4 spaces in your editor, it is not always the case. Better use spaces.

Most editors can be configured so that Tab key is automatically converted into spaces.